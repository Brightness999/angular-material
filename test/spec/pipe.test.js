/**
 * @fileoverview Test the pipe interface.
 */

const Logality = require('../..');
const { stubLogality, cooldown } = require('../lib/tester.lib');

describe('pipe interface', () => {
  stubLogality();
  test('Will pipe one logality instance to another', () => {
    let outputDone = false;
    const logalityChild = new Logality();
    const logalityParent = new Logality({
      output: (logMessage) => {
        expect(logMessage).toMatchSnapshot();
        outputDone = true;
      },
    });

    logalityParent.pipe(logalityChild);
    logalityChild.get().log('hello world');
    expect(outputDone).toBeTrue();
  });

  test('Will pipe multiple logality instances to a single one', () => {
    const logalityChild1 = new Logality();
    const logalityChild2 = new Logality();
    const logalityChild3 = new Logality();

    const mockOut = jest.fn();

    const logalityParent = new Logality({
      output: mockOut,
    });

    logalityParent.pipe([logalityChild1, logalityChild2, logalityChild3]);
    logalityChild1.get().log('hello world 1');
    logalityChild2.get().log('hello world 2');
    logalityChild3.get().log('hello world 3');

    expect(mockOut).toHaveBeenCalledTimes(3);
  });
  test('Will pipe logality instance with custom output and objectMode', () => {
    const logalityChild = new Logality({
      objectMode: true,
      output: (logMessage) => {
        return JSON.stringify(logMessage);
      },
    });

    const output = jest.fn();
    const logalityParent = new Logality({
      output,
    });

    logalityParent.pipe(logalityChild);

    logalityChild.get().log('hello world');

    expect(output).toHaveBeenCalled();
    expect(output).toHaveBeenLastCalledWith('');
  });
  test('Will pipe logality instance with async custom output and objectMode', async () => {
    const logalityChild = new Logality({
      objectMode: true,
      async: true,
      output: async (logMessage) => {
        await cooldown();
        return JSON.stringify(logMessage);
      },
    });

    const output = jest.fn();
    const logalityParent = new Logality({
      output,
    });

    logalityParent.pipe(logalityChild);

    await logalityChild.get().log('hello world');

    expect(output).toHaveBeenCalled();
    expect(output).toHaveBeenLastCalledWith('');
  });
});
