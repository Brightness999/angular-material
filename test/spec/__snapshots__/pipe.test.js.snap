// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pipe interface Parent's middleware will receive the piped child messages 1`] = `
Object {
  "context": Object {
    "runtime": Object {
      "application": "child",
    },
    "source": Object {
      "file_name": "/test/spec/pipe.test.js",
    },
    "system": Object {
      "hostname": "localhost",
      "pid": 36255,
      "process_name": "node .",
    },
  },
  "dt": "2018-05-18T16:25:57.815Z",
  "event": Object {},
  "level": "info",
  "message": "hello world",
  "severity": 6,
}
`;

exports[`pipe interface Piped async child with object custom output will go through synch parent custom output 1`] = `
Object {
  "context": Object {
    "runtime": Object {
      "application": "child",
    },
    "source": Object {
      "file_name": "/test/spec/pipe.test.js",
    },
    "system": Object {
      "hostname": "localhost",
      "pid": 36255,
      "process_name": "node .",
    },
  },
  "dt": "2018-05-18T16:25:57.815Z",
  "event": Object {},
  "level": "info",
  "message": "hello world",
  "severity": 6,
}
`;

exports[`pipe interface Piped async child with object custom output will go through synch parent custom output 2`] = `
"{\\"level\\":\\"info\\",\\"severity\\":6,\\"dt\\":\\"2018-05-18T16:25:57.815Z\\",\\"message\\":\\"hello world\\",\\"context\\":{\\"runtime\\":{\\"application\\":\\"child\\"},\\"source\\":{\\"file_name\\":\\"/test/spec/pipe.test.js\\"},\\"system\\":{\\"hostname\\":\\"localhost\\",\\"pid\\":36255,\\"process_name\\":\\"node .\\"}},\\"event\\":{}}
"
`;

exports[`pipe interface Piped async child with string custom output will go through parent custom output 1`] = `"{\\"level\\":\\"info\\",\\"severity\\":6,\\"dt\\":\\"2018-05-18T16:25:57.815Z\\",\\"message\\":\\"hello loki world\\",\\"context\\":{\\"runtime\\":{\\"application\\":\\"child\\"},\\"source\\":{\\"file_name\\":\\"/test/spec/pipe.test.js\\"},\\"system\\":{\\"hostname\\":\\"localhost\\",\\"pid\\":36255,\\"process_name\\":\\"node .\\"}},\\"event\\":{}}"`;

exports[`pipe interface Will pipe one logality instance to another 1`] = `
Object {
  "context": Object {
    "runtime": Object {
      "application": "child",
    },
    "source": Object {
      "file_name": "/test/spec/pipe.test.js",
    },
    "system": Object {
      "hostname": "localhost",
      "pid": 36255,
      "process_name": "node .",
    },
  },
  "dt": "2018-05-18T16:25:57.815Z",
  "event": Object {},
  "level": "info",
  "message": "hello world",
  "severity": 6,
}
`;

exports[`pipe interface Will pipe three logality instances to another in chain 1`] = `
Object {
  "context": Object {
    "runtime": Object {
      "application": "child",
    },
    "source": Object {
      "file_name": "/test/spec/pipe.test.js",
    },
    "system": Object {
      "hostname": "localhost",
      "pid": 36255,
      "process_name": "node .",
    },
  },
  "dt": "2018-05-18T16:25:57.815Z",
  "event": Object {},
  "level": "info",
  "message": "hello world",
  "severity": 6,
}
`;

exports[`pipe interface Will pipe three logality instances to another in chain 2`] = `
Object {
  "context": Object {
    "runtime": Object {
      "application": "child",
    },
    "source": Object {
      "file_name": "/test/spec/pipe.test.js",
    },
    "system": Object {
      "hostname": "localhost",
      "pid": 36255,
      "process_name": "node .",
    },
  },
  "dt": "2018-05-18T16:25:57.815Z",
  "event": Object {},
  "level": "info",
  "message": "hello world",
  "severity": 6,
}
`;
